### ISSUE TITLE
Beta3 buries number of rows returned in output window

### ISSUE THREAD

**Issue by beRto-** [sqlitebrowser/sqlitebrowser#1709]
Previous DB4S put the number of rows returned and the execution time right at the top of the console window:

`10 rows returned in 0ms from: select * from testdata2`

Beta3 (and nightly) puts it in the last row:

```
Execution finished without errors.

-- At line 1:
select * from testdata2
where field1 like '%1'
-- Result: 1 rows returned in 14ms
```
This means that if you run a long query, you need to scroll all the way down to see what was returned. The query itself is useful, but already known (user just ran it, after all). The number of rows returned and execution time are new information and should be available up front.

Suggested output:

```
Execution finished without errors.
-- Result: 1 rows returned in 14ms

-- At line 1:
select * from testdata2
where field1 like '%1'
```

**Comment by sky5walk** [comment]
[I agree.](https://github.com/sqlitebrowser/sqlitebrowser/issues/1670#issuecomment-449135826) 👍

**Comment by mgrojo** [comment]
The Beta3 does not output "Execution finished without errors", that was added later. But in any case, the problem exist for long queries. But changing that in the beta would change the strings and we should freeze them for the translators, otherwise the translation cycle would never end.

I think that an alternative solution is to scroll the results panel to the end each time the results are output. That would bring visibility to the important part.

**Comment by beRto-** [comment]
@mgrojo - scrolling would technically work, but seems a little awkward. If reorganizing the text is really not feasible, then I suppose I would prefer the scrolling option to the current setup. But I do think laying out the text in the most logical way is the right way to fix this.

Thanks for taking the time to consider this request!

**Comment by sky5walk** [comment]
I would only scroll to the bottom if the total rows affected are not known until the last write to the editor window. Else, keep SUCCESS/FAIL/ERROR and "nRows affected" up on 1st/2nd lines.

**Comment by mgrojo** [comment]
Just a quick hint, remember that you can also reduce the font size in Preferences: "SQL > SQL results font size".

**Comment by beRto-** [comment]
Sorry if this is a silly question, but I'm a little confused. I just installed 3.11.1 and still see the result like at the very end of the query output? The reason I am confused is that I thought a fix for this was included under #1670 which is "checked" in the release notes #1747? Am I missing something?

**Comment by mgrojo** [comment]
> Sorry if this is a silly question, but I'm a little confused. I just installed 3.11.1 and still see the result like `-- Result: 40 rows returned in 82ms` at the very end of the query output (necessitating scrolling on long queries)? The reason I am confused is that I thought a fix for this was included under #1670 which is "checked" in the release notes #1747? Am I missing something?

Only the first enhancement request entered the released, that is, allowing the "bottom editor of Execute SQL tab to grow taller". I've made now some reordering of the output, so the result message and the line are before the query.

@beRto- @sky5walk Could you try with tomorrow's nightly build and give your feedback?

@maratumba You may also like this change, because it allows you to collapse the bottom results panel since you can still check the results in the SQL log with User selection. In this way, you will gain some vertical space in the SQL editor and query result table.

**Comment by sky5walk** [comment]
Yes, that is a step forward. But, why not drop the bottom results window altogether since it is available in the SQL log? Instead, write SUCCESS or FAIL to the result table. While iterating complex queries, the results table is useless as it is always empty. 🤕

**Comment by mgrojo** [comment]
> Yes, that is a step forward. But, why not drop the bottom results window altogether since it is available in the SQL log? Instead, write SUCCESS or FAIL to the result table. While iterating complex queries, the results table is useless as it is always empty. face_with_head_bandage

Maybe at some time, but it'd require more time and effort.

**Comment by beRto-** [comment]
@mgrojo - nice! I just tried out `Version 3.11.99 (Feb 24 2019)` and confirm the results status output shows up at the top, with the (long) query following afterwards. Thanks :)

**Comment by chrisjlocke** [comment]
@berto- So this can be closed now?

**Comment by beRto-** [comment]
In terms of development, I agree the proposed fix looked good and the ticket can be closed. That said, I am not sure if / when the fix will go into a release - I inquired in #1773 about adding it to 3.11.2

**Comment by mgrojo** [comment]
@beRto- It will be present in v3.11.2. I guess this can be finally closed :smile:
